lib/main.js

```diff
@@ -327,6 +371,22 @@ function signText(text) {
       return ERROR_INTERNAL;
     }
 
+    slotList = PK11_GetAllSlotsForCert(cert, null);
+    if (slotList.isNull()) {
+      log("PK11_GetAllSlotsForCert failed");
+      cleanupSignTextResources(cert, contentInfo, encodedItem, slotList);
+      return ERROR_INTERNAL;
+    }
+    let it = slotList.contents.head;
+    while (it && !it.isNull()) {
+      if (PK11_CheckUserPassword(it.contents.slot, certName.password) != SECSuccess) {
```

Comment:
I don't agree on a). Looking to old code I think that purpose of password field is to make sure that password is entered by user. If certificate is on smart card and the user used it before in this browser session, then walked away without unpluging the card, the card session will be open and PK11_IsLoggedIn will return PRTrue. This means that other people will be able to use certificate for signing even without password. PK11_CheckUserPassword checks if slot needs login so call to PK11_NeedLogin is not needed, but will I'll add it just to be safe.

About point b, you are right. Although looking to CERTCertificateStr structure I don't think it is currently possible for certificate to be on more than one slot, maybe some day it will be and I will change the code to check for SEC_ERROR_BAD_PASSWORD.